src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -309,6 +309,28 @@ public void setFieldNameToClassNameMap(
     return classAndItsParent.values();
   }
 
+  /**
+   * Get the names of members that could be used by the target methods.
+   *
+   * @return the value of potentialUsedMembers.
+   */
+  public Set<String> getPotentialUsedMembers() {
+    Set<String> copyOfPotentialUsedMembers = new HashSet<>();
+    copyOfPotentialUsedMembers.addAll(potentialUsedMembers);
+    return copyOfPotentialUsedMembers;
+  }
+
+  /**
+   * Get the list of synthetic classes created by UnsolvedSymbolVisitor.
+   *
+   * @return the value of missingClass.
+   */
+  public Set<UnsolvedClassOrInterface> getMissingClass() {
+    Set<UnsolvedClassOrInterface> copyOfMissingClass = new HashSet<>();
+    copyOfMissingClass.addAll(missingClass);
```

Comment:
Can an `UnsolvedClassOrInterface` object be modified in a later run of `UnsolvedSymbolVisitor`? I think that the answer is yes: new methods can be added, the number of type variables might be updated, etc.

In that case, this copy operation isn't sufficient: you're creating a new set, yes, but the new set still contains the same set of pointers as the old set. So any change _to those objects_ will still be reflected in the new set.

(We may also need to implement a copy constructor for `UnsolvedClassOrInterface`.)

Note after I read the rest of the PR: my concern here is valid in general, but in this specific case the code is already correct, because this copy is transformed into Strings (which are immutable) immediately. However, that's not obvious from the way that the code is written, so I think we should make a change here to make it clear why this is safe despite `UnsolvedClassOrInterface` being mutable: in particular, I suggest that `getStringSetFromSyntheticClassSet` be called on `copyOfMissingClass` here before the result is returned (and the return type of this method (and its name, Javadoc, etc) be changed to `Set<String>`.